adapters/repos/db/vector/cuvs/index.go

```diff
@@ -0,0 +1,895 @@
+			return nil, err
+		}
+
+		cuvsSearchParams.SetAlgo(cagra.SearchAlgoMultiCta)
+		cuvsSearchParams.SetItopkSize(256)
+		cuvsSearchParams.SetSearchWidth(1)
+
+		cfg.CuvsSearchParams = cuvsSearchParams
+
+	}
+
+	if err != nil {
+		return nil, err
+	}
+
+	cuvsIndex, err := cagra.CreateIndex()
+	if err != nil {
+		return nil, fmt.Errorf("create cuvs index: %w", err)
+	}
+
+	internals := cuvs_internals{
+		cuvsIndex:        cuvsIndex,
+		cuvsIndexParams:  cfg.CuvsIndexParams,
+		cuvsSearchParams: cfg.CuvsSearchParams,
+		cuvsResource:     &res,
+		dlpackTensor:     nil,
+	}
+
+	index := &cuvs_index{
+		cuvs_internals: internals,
+		id:             cfg.ID,
+		targetVector:   cfg.TargetVector,
+		logger:         logger,
+		distanceMetric: cuvs.DistanceL2,
+		store:          store,
+		rootPath:       cfg.RootPath,
+
+		idCuvsIdMap:    *NewBiMap(),
+		cuvsPoolMemory: cfg.CuvsPoolMemory,
+	}
+
+	index.maxFiltered = 30
+	index.nextCuvsId = 0
+
+	if err := index.initBuckets(context.Background()); err != nil {
+		return nil, fmt.Errorf("init cuvs index buckets: %w", err)
+	}
+
+	if err := index.initMetadata(); err != nil {
+		return nil, err
+	}
+
+	return index, nil
+}
+
+func byteSliceFromFloat32Slice(vector []float32, slice []byte) []byte {
+	for i := range vector {
+		binary.LittleEndian.PutUint32(slice[i*4:], math.Float32bits(vector[i]))
+	}
+	return slice
+}
+
+func ValidateUserConfigUpdate(initial, updated schemaConfig.VectorIndexConfig) error {
+	_, ok := initial.(cuvsEnt.UserConfig)
+	if !ok {
+		return errors.Errorf("initial is not UserConfig, but %T", initial)
+	}
+
+	_, ok = updated.(cuvsEnt.UserConfig)
+	if !ok {
+		return errors.Errorf("updated is not UserConfig, but %T", updated)
+	}
+
+	return nil
+}
+
+func (index *cuvs_index) AddMulti(ctx context.Context, docId uint64, vector [][]float32) error {
+	return errors.Errorf("AddMulti is not supported for cuvs index")
+}
+
+func (index *cuvs_index) AddMultiBatch(ctx context.Context, docIds []uint64, vectors [][][]float32) error {
+	return errors.Errorf("AddMultiBatch is not supported for cuvs index")
+}
+
+func (index *cuvs_index) DeleteMulti(id ...uint64) error {
+	return errors.Errorf("DeleteMulti is not supported for cuvs index")
+}
+
+func (index *cuvs_index) SearchByMultiVector(ctx context.Context, vector [][]float32, k int, allow helpers.AllowList) ([]uint64, []float32, error) {
+	return nil, nil, errors.Errorf("SearchByMultiVector is not supported for cuvs index")
+}
+
+func (index *cuvs_index) ValidateMultiBeforeInsert(vector [][]float32) error {
+	return errors.Errorf("ValidateMultiBeforeInsert is not supported for cuvs index")
+}
+
+func (index *cuvs_index) GetKeys(id uint64) (uint64, uint64, error) {
+	return 0, 0, errors.Errorf("GetKeys is not supported for cuvs index")
+}
+
+func (index *cuvs_index) Multivector() bool {
+	return false
+}
+
+func shouldExtend(index *cuvs_index, num_new uint64) bool {
+	if num_new > 100 {
+		println("num_new is greater than 100; rebuilding")
```

Comment:
All the `println` statements should be replaced with `logger.Debugf` and `logger.Infof` as required. Please set `action=cuvs` in the logs.